hydra_agent/agent.py

```diff
@@ -41,26 +41,41 @@ def initialize_graph(self) -> None:
         self.redis_connection.sadd("fs:url", self.entrypoint_url)
 
     def get(self, url: str = None, resource_type: str = None,
-            filters: dict = {}) -> Union[dict, list]:
+            filters: dict = {},
+            cached_limit: int = sys.maxsize) -> Union[dict, list]:
```

Comment:
Not sure if using `sys.maxsize` as the cached_limit is a good idea here.
From StackOverflow

> Python can handle arbitrarily large integers in computation. Any integer too big to fit in 64 bits (or whatever the underlying hardware limit is) is handled in software. For that reason, Python 3 doesn't have a sys.maxint constant.
The value sys.maxsize, on the other hand, reports the platform's pointer size, and that limits the size of Python's data structures such as strings and lists.

What are we trying to achieve with this?

Reply:
This was a way to use the Redis cached data for collections and search by type/classes, so the user wouldn't always query the server. Maxsize is a way to set that by default it will always fetch the server because Redis won't have "Maxsize" amount of nodes, which then allows the user to control by itself.

Now that we have https://github.com/HTTP-APIs/hydrus/pull/409 I could check for the pagination limit and return only the cached data if enough, but if the user is querying multiple pages this might lead to inconsistency problems because the cached data won't be in the same order.

Removing this functionality is also an option, then Redis use case will be mainly on getting specific collection members and its properties instead of going for the server.

Reply:
Let's go with your implementation @Guttz 
we can double-check later eventually.